### PR TITLE
Mock Credentials in AWSPrometheusRemoteWriteExporter Tests

### PR DESCRIPTION
**Pull request by JasonXZLiu**
**Description:**

We are adding more tests to the AWSPrometheusRemoteWriteExporter in this PR. These tests include mocking the credentials and test for negative error cases. In this PR, we refactored `auth.go` so that we can provide mocked credentials to the SigV4 signer and added a test for no credentials.

**Testing:** 

* `auth_test.go` was changed to provide mock credentials
* `factory_test.go` was changed to add a comment to warn user's not to set their credentials with environment variables in production code

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -69,6 +69,8 @@ require (
 // Replace references to modules that are in this repository with their relateive paths
 // so that we always build with current (latest) version of the source code.
 
+replace go.opentelemetry.io/collector => ../opentelemetry-collector
```
Comment (JasonXZLiu) [inline go.mod:72]: Remove this from the PR

**Review by alolita — APPROVED** [review]
